code/modules/clothing/head/hardhat.dm

```diff
@@ -124,9 +124,9 @@
 	tint = 2
 	flags_inv = HIDEEYES | HIDEFACE | HIDESNOUT
 	flags_cover = HEADCOVERSEYES | HEADCOVERSMOUTH
-	visor_vars_to_toggle = VISOR_FLASHPROTECT | VISOR_TINT
+	visor_vars_to_toggle = VISOR_FLASHPROTECT | VISOR_TINT | PEPPERPROOF
```

Comment:
You definitely can't use the same `PEPPERPROOF` flag in `visor_vars_to_toggle`, you would need to define a new `VISOR_PEPPERPROTECT`